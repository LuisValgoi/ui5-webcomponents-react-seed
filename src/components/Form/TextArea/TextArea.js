import React from 'react';

import { spacing } from '@ui5/webcomponents-react-base';
import { TextArea as UI5TextArea, ValueState } from '@ui5/webcomponents-react';
import FieldBase from '../FieldBase/FieldBase';

const TextArea = ({ field, form: { touched, errors }, labelText, rows, style, ...props }) => {
  const errorMsg = touched[field.name] && errors[field.name];
  const errorState = errorMsg ? ValueState.Error : ValueState.None;

  return (
    <FieldBase labelText={labelText} required={props.required} for={props.for} showColon={props.showColon}>
      <UI5TextArea
        data-testid="textArea-wrapper"
        valueState={errorState}
        valueStateMessage={<span>{errorMsg}</span>}
        rows={rows}
        style={style ? style : spacing.sapUiSmallMarginBottom}
        {...props}
        {...field}
      />
    </FieldBase>
  );
};

export default TextArea;
