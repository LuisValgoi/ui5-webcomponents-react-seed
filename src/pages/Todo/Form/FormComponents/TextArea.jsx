import React from 'react';

// styling
import { spacing } from '@ui5/webcomponents-react-base';

//components
import { TextArea as UI5TextArea, ValueState, FlexBox, FlexBoxDirection } from '@ui5/webcomponents-react';
import Label from './Label';

const TextArea = ({ field, form: { touched, errors }, labelText, rows, ...props }) => {
  const errorMsg = touched[field.name] && errors[field.name];
  const errorState = errorMsg ? ValueState.Error : ValueState.None;
  return (
    <FlexBox direction={FlexBoxDirection.Column}>
      <Label>{labelText}</Label>
      <UI5TextArea valueState={errorState} valueStateMessage={<span>{errorMsg}</span>} style={spacing.sapUiSmallMarginBottom} rows={rows} {...props} {...field} />
    </FlexBox>
  );
};

export default TextArea;
