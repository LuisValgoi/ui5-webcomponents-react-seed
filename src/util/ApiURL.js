export default {
  GET_USER_LOGGED: '/v1/user/logged',
  GET_TODO_LIST: '/v1/todo/all',
  GET_TODO_BY_ID: '/v1/todo/:id'
};
