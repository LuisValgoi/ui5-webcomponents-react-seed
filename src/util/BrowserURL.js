export default {
  HOME: '/',
  TODO_LIST: '/todo/list',
  TODO_DETAIL: '/todo/:id/detail',
  NOT_FOUND: '/notFound',
  BUGGY: '/buggy',
  ANY: '/*'
};
