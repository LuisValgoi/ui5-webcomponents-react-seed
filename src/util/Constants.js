export default {
  REACT_QUERY: {
    CODES: {
      SUCCESS: 'success',
      LOADING: 'loading',
      ERROR: 'error',
    },
    KEYS: {
      RQ_GET_USER_LOGGED: 'RQ_GET_USER_LOGGED',
      RQ_GET_TODO_LIST: 'RQ_GET_TODO_LIST',
      GET_TODO_BY_ID: 'GET_TODO_BY_ID',
    },
  },
  SEED: {
    SELECTED_THEME: 'SELECTED_THEME',
  },
};
