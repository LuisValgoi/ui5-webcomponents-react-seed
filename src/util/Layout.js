import React from 'react';
import { Text } from '@ui5/webcomponents-react/lib/Text';

export const TextWrapper = ({ children, isOnHeader, style }) => {
  return (
    <Text
      tooltip={children}
      wrapping
      className={isOnHeader && 'text-header-color'}
      style={{ ...style, lineHeight: '20px' }}
    >
      {children}
    </Text>
  );
};
