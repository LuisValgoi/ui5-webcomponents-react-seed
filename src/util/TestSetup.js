import React from 'react';

import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

import BrowserProvider from './browser/BrowserProvider';

const render = (ui, { route = BrowserProvider.getUrl('HOME'), ...renderOptions } = {}) => {
  const WrapperProvider = ({ ...props }) => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return (
      <BrowserRouter>
        <Router history={history}>{props.children}</Router>
      </BrowserRouter>
    );
  };

  return rtlRender(ui, { wrapper: WrapperProvider, ...renderOptions });
};

const serverCustom = (apiUrl, data) => {
  return setupServer(
    rest.get(`*${apiUrl}`, (req, res, ctx) => {
      return res(ctx.json(data));
    }),
  );
};

export * from '@testing-library/react';

export { render, serverCustom };
