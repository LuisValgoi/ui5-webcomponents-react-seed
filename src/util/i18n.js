import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import LanguageDetector from 'i18next-browser-languagedetector';

import translationEN from '../locales/en/translation.json';
import translationPT from '../locales/pt/translation.json';

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    debug: true,
    keySeparator: false,
    interpolation: {
      escapeValue: false,
    },
    resources: {
      en: {
        translation: translationEN,
      },
      pt: {
        translation: translationPT,
      },
    },
    react: {
      useSuspense: false,
    },
  });

export default i18n;
