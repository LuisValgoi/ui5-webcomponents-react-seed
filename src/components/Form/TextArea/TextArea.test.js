import React from 'react';

import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../../util/TestSetup';

import TextArea from '../TextArea/TextArea';

describe('TextArea.js Test Suite', () => {
  test('should match snapshot', () => {
    const { asFragment } = render(
      <TextArea labelText="Text Content" name="description" field={{ name: 'description' }} form={{ touched: { name: 'description' }, errors: { name: 'description' } }} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('should have rendered with the fieldbase', () => {
    render(<TextArea labelText="Text Content" name="description" field={{ name: 'description' }} form={{ touched: { name: 'description' }, errors: { name: 'description' } }} />);

    const fieldbase = screen.getByTestId('fieldbase-wrapper');
    const component = screen.getByTestId('textArea-wrapper');

    expect(fieldbase).toBeInTheDocument();
    expect(component).toBeInTheDocument();
  });
});
