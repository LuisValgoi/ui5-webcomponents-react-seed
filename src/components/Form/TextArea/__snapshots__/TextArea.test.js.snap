// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea.js Test Suite should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="FlexBox-flexBox-0-2-1 FlexBox-flexBoxDirectionColumn-0-2-8 FlexBox-justifyContentStart-0-2-3 FlexBox-alignItemsStretch-0-2-15 FlexBox-flexWrapNoWrap-0-2-17"
    data-testid="fieldbase-wrapper"
  >
    Label {
      "_domRefReadyPromise": Promise {
        "_deferredResolve": [Function],
      },
      "_firePropertyChange": false,
      "_fullyConnected": false,
      "_inDOM": false,
      "_monitoredChildProps": Map {},
      "_shouldInvalidateParent": false,
      "_state": Object {
        "for": "",
        "required": false,
        "showColon": false,
        "wrap": true,
      },
      "_upToDate": false,
    }
    <ui5-textarea
      data-testid="textArea-wrapper"
      growing-max-lines="0"
      name="description"
      rows="0"
      style="margin-bottom: 1rem;"
      value-state="None"
    >
      <span
        slot="valueStateMessage"
      />
    </ui5-textarea>
  </div>
</DocumentFragment>
`;
